app/models/course/assessment/question/programming.rb

```diff
@@ -173,7 +173,7 @@ def process_package # rubocop:disable Metrics/CyclomaticComplexity, Metrics/Perc
 
   def should_evaluate_package
     time_limit_changed? || memory_limit_changed? ||
-    language_id_changed? || (is_codaveri_changed? && is_codaveri) ||
```

Comment:
Seems like this restriction was added with 8b144c897ac3caeaa20d423c826422b9a3e5b34e in  https://github.com/Coursemology/coursemology2/pull/4867. Seems like the addition of `is_codaveri` is to make sure the package is evaluated if also the evaluator is set to Codaveri? Not sure I understand the original intention for this condition.

Either way, doesn't seem like this line has anything to do with the ability to switch evaluators when submissions exist, does it?

If indeed switching evaluators have nothing to do with the presence of submissions, and just has to do with allowing it on the client form, why do we have this restriction in the first place?

Reply:
Yes you are right - the package is evaluated + any existing answers are re-evaluated.

Initially, we only allow any change from codaveri to default if there is no submission -> no package evaluation + no re-evaluation.

Now, we re-enable it to mainly allow re-evaluation.

Reply:
Because it was possible to switch from codaveri to default when there's no submission -> no package reeval is needed